tests/LithologicalDescriptionControllerTest.cs

```diff
@@ -0,0 +1,206 @@
+        Assert.AreEqual("Metrics Mountains Practical", lithologicalDescription.Description);
+        Assert.AreEqual(9000, lithologicalDescription.QtDescriptionId);
+        Assert.AreEqual(6_000_001, lithologicalDescription.StratigraphyId);
+    }
+
+    [TestMethod]
+    public async Task EditLithologicalDescriptionWithCompleteLithologicalDescription()
+    {
+        var id = 9_000_069;
+        var originalLithologicalDescription = new LithologicalDescription
+        {
+            CreatedBy = null,
+            CreatedById = 1,
+            Created = new DateTime(2021, 11, 30, 12, 10, 33).ToUniversalTime(),
+            Description = "Libyan Dinar 1080p leading edge",
+            FromDepth = 90,
+            ToDepth = 100,
+            Id = 9_000_069,
+            IsLast = true,
+            QtDescription = null,
+            QtDescriptionId = 9001,
+            Updated = new DateTime(2021, 3, 31, 16, 55, 02).ToUniversalTime(),
+            UpdatedBy = null,
+            UpdatedById = 5,
+            StratigraphyId = 6_000_006,
+        };
+
+        var newLithologicalDescription = new LithologicalDescription
+        {
+            Id = id,
+            CreatedById = 4,
+            UpdatedById = 4,
+            Created = new DateTime(2021, 2, 14, 8, 55, 34).ToUniversalTime(),
+            StratigraphyId = 6_000_010,
+            Description = "Freddy ate more cucumber than Maria.",
+            QtDescriptionId = 9003,
+        };
+
+        var lithologicalDescriptionToEdit = context.LithologicalDescriptions.Single(c => c.Id == id);
+        Assert.AreEqual(1, lithologicalDescriptionToEdit.CreatedById);
+        Assert.AreEqual(5, lithologicalDescriptionToEdit.UpdatedById);
+        Assert.AreEqual(6_000_006, lithologicalDescriptionToEdit.StratigraphyId);
+        Assert.AreEqual("Libyan Dinar 1080p leading edge", lithologicalDescriptionToEdit.Description);
+
+        // Update LithologicalDescription
+        var response = await controller.EditAsync(newLithologicalDescription);
+        var okResult = response as OkObjectResult;
+        Assert.AreEqual(200, okResult.StatusCode);
+
+        // Assert Updates and unchanged values
+        var updatedContext = ContextFactory.CreateContext();
+        var updatedLithologicalDescription = updatedContext.LithologicalDescriptions.Single(c => c.Id == id);
+
+        Assert.AreEqual(4, updatedLithologicalDescription.CreatedById);
+        Assert.AreEqual(1, updatedLithologicalDescription.UpdatedById);
+        Assert.AreEqual(6_000_010, updatedLithologicalDescription.StratigraphyId);
+        Assert.AreEqual("Freddy ate more cucumber than Maria.", updatedLithologicalDescription.Description);
+        Assert.AreEqual(9003, updatedLithologicalDescription.QtDescriptionId);
+
+        // Reset edits
+        var cleanupContext = ContextFactory.CreateContext();
+        cleanupContext.Update(originalLithologicalDescription);
+        await cleanupContext.SaveChangesAsync();
```

Comment:
PP: Den Reset in ein `try-finally`-Block packen, so dass der Reset auch im Fehlerfall ausgeführt wird.